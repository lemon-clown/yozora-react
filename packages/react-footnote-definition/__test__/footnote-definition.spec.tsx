import { render } from 'enzyme'
import React from 'react'
import FootnoteDefinition from '../src'
import type { IFootnoteDefinitionProps } from '../src'

const footnoteDefinitionProps: IFootnoteDefinitionProps = {
  label: '1',
  identifier: 'footnote-1',
  children: <span>waw1</span>,
}

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  describe('className is optional', function () {
    it('default', function () {
      const node = render(<FootnoteDefinition {...footnoteDefinitionProps} />)
      expect(node.hasClass('yozora-footnote-definition')).toBeTruthy()
    })

    it('custom', function () {
      const node = render(
        <FootnoteDefinition {...footnoteDefinitionProps} className="my-footnote-definition" />,
      )
      expect(node.hasClass('yozora-footnote-definition')).toBeTruthy()
      expect(node.hasClass('my-footnote-definition')).toBeTruthy()
    })
  })

  it('style is optional', function () {
    const node = render(
      <FootnoteDefinition {...footnoteDefinitionProps} style={{ color: 'orange' }} />,
    )
    expect(node.css('color')).toEqual('orange')
  })
})

describe('snapshot', function () {
  it('basic', function () {
    const wrapper = render(
      <FootnoteDefinition
        {...footnoteDefinitionProps}
        className="custom-footnote-definition"
        style={{ marginTop: '2rem' }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
