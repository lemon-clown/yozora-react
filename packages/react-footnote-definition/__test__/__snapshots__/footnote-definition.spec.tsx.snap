// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `
<li
  class="yozora-footnote-definition custom-footnote-definition"
  id="footnote-1"
  style="margin-top:2rem"
>
  <p
    class="yozora-footnote-definition__title yozora-paragraph"
  >
    <a
      href="#reference-footnote-1"
    >
      â†‘
    </a>
    <span>
      1
    </span>
  </p>
  <div
    class="yozora-footnote-definition__content"
  >
    <span>
      waw1
    </span>
  </div>
</li>
`;
